src/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -65,6 +65,8 @@ internal ZipArchiveEntry(ZipArchive archive, ZipCentralDirectoryFileHeader cd)
             _originallyInArchive = true;
 
             _diskNumberStart = cd.DiskNumberStart;
+            _versionMadeByPlatform = (ZipVersionMadeByPlatform)cd.VersionMadeByCompatibility;
```

Comment:
Good catch! The value written out will be the value that we initialized it to in the constructor. So if we have a zip with OS version "13" and read it on a Windows system and then write it again the written zip will have the OS set to "13".

The OS tag is <i>supposed</i> to be set to the system that the zip was written on, so this behavior is incorrect. It's possible that it would be okay in the scenario where the zip is opened and rewritten without modification, but i could see that causing more issues than it would circumvent.

I'm going to modify the write code to always write the CurrentZipPlatform.